### PR TITLE
💅🏾 Material You implementation!

### PR DESCRIPTION
**Pull request by ElyasAsmad**
# Material You implementation

## Description
Overhauled Home Screen design and implemented Material You across all screens inside the app

Refer #30 

## Type of Change
- Home Screen redesign.
- Material You implementation based on [Google's Material Design 3](https://m3.material.io/) principle. 

## Update Preview

https://user-images.githubusercontent.com/78308067/198932000-15604553-33cb-4906-a3ea-2623e4fcc3d4.mp4

## More Information
Device: OnePlus 9 (LE2115)
OS: OxygenOS 12 (Android 12)
App Version: 1.0.3

Tested by me and @HaikalAkif

## Side Note
I had a lot of free time last weekend and tried to redesign the User Interface of this app

### CODE REVIEW

**Comment by iqfareez** [comment]
Hi. May I ask which Flutter channel you are on? Because in `stable`, some changes in [material_color_utilities](https://pub.dev/packages/material_color_utilities/changelog#020) are not available yet.

**Comment by iqfareez** [comment]
image

One minor issue though, the cards in validator page is dark when device in light mode.

**Review by iqfareez — APPROVED** [review]
Looks good. Thank you.

**Comment by ElyasAsmad** [comment]
> Looks good. Thank you.

I'm happy to contribute to your project and to the IIUM Community!


> Also I've been wondering how do you generate the `color_schemes.g.dart` file?

I generated the color_schemes file using Google's online version of their [Material Theme Builder](https://m3.material.io/theme-builder#/custom). Anyways, I am still finding ways on how to use the user's current theme instead of choosing only one color for the Material 3 theme version of the app.

**Comment by ElyasAsmad** [comment]
For the Material You implementation, I have successfully configure the app to use user's dynamic colour palette instead of using the hardcoded colour!

Refer https://github.com/iqfareez/iium_schedule/commit/e64a13c2924e66bc5cab8b32b96fac887f96b05d

File: lib/util/lane_events_util.dart [hunk 1]
```diff
@@ -71,13 +71,16 @@ class LaneEventsUtil {
               : Colors.white;
 
           return TableEvent(
+            decoration: BoxDecoration(
+              borderRadius: const BorderRadius.all(Radius.circular(15.0)),
```
Comment (iqfareez) [inline lib/util/lane_events_util.dart:75]: Nice one 👍 . Previously the border is too rounded.

File: lib/main.dart [hunk 2]
```diff
@@ -41,23 +42,39 @@ class MyApp extends StatelessWidget {
...
-          ),
-        ),
-        themeMode: ThemeMode.system,
-        home: const MyBody(),
+      child: DynamicColorBuilder(
+        builder: (ColorScheme? lightColorScheme, ColorScheme? darkColorScheme) {
+          return MaterialApp(
+            title: 'IIUM Schedule',
+            theme: ThemeData(
+              colorScheme: lightColorScheme,
```
Comment (iqfareez) [inline lib/main.dart:50]:
```suggestion
              colorScheme: lightColorScheme ?? const ColorScheme.light(),
```

However, it is not a one liner fix. but just enough as a fallback colour scheme.

File: lib/views/scheduler/input_scope.dart [hunk 3]
```diff
@@ -1,3 +1,5 @@
+import 'dart:ffi';
+
```
Comment (iqfareez) [inline lib/views/scheduler/input_scope.dart:2]:
```suggestion
```
Unused import. Can be removed as it caused compile error on web 😬

**Review by iqfareez — COMMENTED** [review]
Greattt!

I'm not aware that [dynamic_colour](https://pub.dev/packages/dynamic_color) exists until today.

Tested on Android 12 & Windows. it's perfect. ✅ 

However, on Android <12 & Web, some amendments may be needed as dynamic colour is not supported there. Please see suggestions.

flutter_01


<sub>Btw, just ignore the failing checks. hehe.</sub>

File: lib/main.dart [hunk 4]
```diff
@@ -41,23 +42,39 @@ class MyApp extends StatelessWidget {
...
+              //       foregroundColor: Colors.purple.shade200),
+              // ),
+              useMaterial3: true,
+              textTheme: ThemeData.dark().textTheme.apply(
+                fontFamily: 'Inter'
+              ),
+              primaryTextTheme: ThemeData.dark().textTheme.apply(
+                fontFamily: 'Inter'
+              ),
+              colorScheme: darkColorScheme
```
Comment (iqfareez) [inline lib/main.dart:72]: Oops, missed this one.

```suggestion
              colorScheme: darkColorScheme ?? const ColorScheme.dark()),
```

File: lib/util/lane_events_util.dart (hunk 1 above)
Comment (ElyasAsmad) [reply lib/util/lane_events_util.dart:75]: Thanks!

**Pull request merged**